src/modules/edc-demo/pages/catalog-browser/catalog-browser.component.ts

```diff
@@ -28,14 +31,31 @@ export class CatalogBrowserComponent implements OnInit {
   runningNegotiations: Map<string, NegotiationResult> = new Map<string, NegotiationResult>(); // contractOfferId, NegotiationResult
   finishedNegotiations: Map<string, ContractNegotiationDto> = new Map<string, ContractNegotiationDto>(); // contractOfferId, contractAgreementId
   private fetch$ = new BehaviorSubject<SearchParams>({label: '', location: ''});
+  public url: string = "";
+  public ownAssets: boolean = true;
 
   constructor(private apiService: CatalogBrowserService,
               public dialog: MatDialog,
-              public titleService: Title) {
+              public titleService: Title,
+              private authenticationService: AuthenticationService,
+              private route: ActivatedRoute) {
   }
 
   ngOnInit(): void {
-    this.filteredContractOffers$ = this.apiService.getContractOffers();
+    this.authenticationService.userProfile$.subscribe(userProfile => {
+      if (!userProfile) {
+        throw new Error('UserProfile is null or undefined.');
+      }
+      this.url = userProfile.url;
+      this.route.data.subscribe(data => {
+        if (data.title === 'Meine Assets') {
+          this.filteredContractOffers$ = this.apiService.getContractOffers(this.url, this.ownAssets);
```

Comment:
Die Zeile steht identisch in beiden if-Bedingungen. Bei der bool-Variante kann sie alse einfach vor das if (da sie eh ausgefuehrt wird), bei nem enum kann sie in beiden branches stehen bleiben.